checks.d/zk.py

```diff
@@ -96,6 +166,44 @@ def check(self, instance):
                 self.service_check('zookeeper.mode', status, message=message,
                                    tags=sc_tags)
 
+        if zk_version and LooseVersion(zk_version) > LooseVersion("3.4.0"):
+            try:
+                mntr_out = self._send_command('mntr', *cx_args)
+            except ZKConnectionFailure:
+                self.increment('zookeeper.timeouts')
+                self.report_instance_mode(hostname, 'down', tags)
+                raise
+            except:
+                e = sys.exc_info()[1]
+                print >> sys.stderr, "Error: %s" % e
+                self.increment('zookeeper.datadog_client_exception')
+                self.report_instance_mode(hostname, 'unknown', tags)
+                raise
+            else:
+                metrics, mode = self.parse_mntr(mntr_out)
+                mode_tag = "mode:%s" % mode
+                if mode != 'inactive':
+                    for name in metrics:
+                        self.gauge(name, metrics[name], tags=tags + [mode_tag])
+
+
+    def report_instance_mode(self, hostname, mode, tags):
```

Comment:
Now that I understand a little better the intent of `report_instance_mode` it seems that the method of passing in `hostname` (both your current one, and my suggestion) won't really cut it. Since the check is running off a single agent, even when there are multiple ZK instances, the hostnames passed in to this function will be identical every time. Is that we want? The `zookeeper.instances` set will only ever have a size of 1. Instead the `hostname` passed in to this function should be derived from the instance configuration (maybe just `instance.get('host')`)